src/qibo/_openqasm.py

```diff
@@ -0,0 +1,287 @@
+"""Qibo wrapper for QASM 3.0 parser."""
+
+from itertools import repeat
+from typing import Union
+
+import numpy as np
+import openqasm3
+
+import qibo
+from qibo.config import raise_error
+from qibo.gates import FusedGate
+
+
+class DefinedGate:
```

Comment:
I think `CustomGate` or `CustomQASMGate` are better (as in more straightforward) names, but it's fine if you disagree.